src/com/amazon/ion/benchmark/WriteRandomIonValues.java

```diff
@@ -299,37 +301,48 @@ public static void writeRequestedSizeFile(int size, IonWriter writer, File file,
      */
     private static void writeDataToFile(IonWriter writer, IonStruct constraintStruct) throws Exception {
         IonType type = IonType.valueOf(constraintStruct.get(IonSchemaUtilities.KEYWORD_TYPE).toString().toUpperCase());
-        switch (type) {
-            case FLOAT:
-            case SYMBOL:
-                //Not available.
-                break;
-            case INT:
-                writer.writeInt(WriteRandomIonValues.constructInt());
-                break;
-            case STRING:
-                writer.writeString(WriteRandomIonValues.constructString(constraintStruct));
-                break;
-            case DECIMAL:
-                writer.writeDecimal(WriteRandomIonValues.constructDecimal(constraintStruct));
-                break;
-            case TIMESTAMP:
-                writer.writeTimestamp(WriteRandomIonValues.constructTimestamp(constraintStruct));
-                break;
-            case STRUCT:
-                WriteRandomIonValues.constructAndWriteIonStruct(constraintStruct, writer);
-                break;
-            case LIST:
-                WriteRandomIonValues.constructAndWriteIonList(writer, constraintStruct);
-                break;
-            case BLOB:
-                writer.writeBlob(WriteRandomIonValues.constructLobs(constraintStruct));
-                break;
-            case CLOB:
-                writer.writeClob(WriteRandomIonValues.constructLobs(constraintStruct));
-                break;
-            default:
-                throw new IllegalStateException(type + " is not supported.");
+        // Check whether the valid_value constraints provided in the top level constraint struct. If a list of valid_value provided, the generated
+        // value should be selected randomly from the provided valid value list every iteration.
+        // Constraint 'valid_value' has two formats, with annotation '' and without annotation ''. This step only check the format that without annotation.
+        // If the annotation 'range' has been detected, it will be processed in the constructing data steps.
+        if (constraintStruct != null && IonSchemaUtilities.parseValidValue(constraintStruct) != null && IonSchemaUtilities.checkValidRange(constraintStruct) == null) {
+            IonValue value = IonSchemaUtilities.parseValidValue(constraintStruct);
```

Comment:
Consider restructuring this to avoid having to call `parseValidValue` twice.

Reply:
> Consider restructuring this to avoid having to call `parseValidValue` twice.

Updated in the following commit.